server/tests/e2e/activities.e2e-spec.ts

```diff
@@ -0,0 +1,200 @@
+                            }
+                        ]
+                    }
+                },
+                stats: {
+                    create: {
+                        reviews: 1
+                    }
+                }
+            },
+            include: {
+                info: true,
+                credits: true,
+                images: true,
+                stats: true,
+                tracks: {
+                    include: {
+                        zones: true
+                    }
+                }
+            }
+        })
+
+        activity1 = await prisma.activity.create({
+            data: {
+                userID: user1.id,
+                data: 122,
+                type: ActivityTypes.MAP_APPROVED
+            }
+        })
+
+        activity2 = await prisma.activity.create({
+            data: {
+                userID: user1.id,
+                data: 123,
+                type: ActivityTypes.WR_ACHIEVED
+            }
+        })
+
+        activity3 = await prisma.activity.create({
+            data: {
+                userID: user2.id,
+                data: 124,
+                type: ActivityTypes.REVIEW_MADE
+            }
+        })
+
+        const authService = global.auth as AuthService;
+        global.accessToken = (await authService.login(user1)).access_token;
+    });
+
+    describe('GET /api/v1/activities', () => {
+        const expects = (res: request.Response) => expect(res.body).toBeValidPagedDto(ActivityDto);
+        it('should respond with an array of activities', async () => {
+            const res = await get('activities', 200);
+
+            expects(res);
+            expect(res.body.totalCount).toBe(3);
+            expect(res.body.response[0].userID).toBe(user1.id);
+            expect(res.body.response[1].userID).toBe(user1.id);
+            expect(res.body.response[2].userID).toBe(user2.id);
+            expect(res.body.response[2].type).toBe(ActivityTypes.REVIEW_MADE);
+        });
+
+        it('should respond with array of activities with userID parameter', async () => {
+            const res = await get('activities', 200, {userID: user1.id});
+
+            expect(res);
+            expect(res.body.totalCount).toBe(2);
+            expect(res.body.response[1].userID).toBe(user1.id);
+        });
+
+        it('should respond with array of activities with data parameter', async () => {
+            const res = await get('activities', 200, {data: activity3.data});
+
+            expect(res);
+            expect(res.body.totalCount).toBe(1);
+            expect(res.body.response[0].userID).toBe(user2.id);
+            expect(res.body.response[0].data).toBe('124');
+        });
+
+        it('should respond with array of activities with type paramater', async () => {
+            const res = await get('activities', 200, {type: 'MAP_APPROVED'});
+
+            expect(res);
+            expect(res.body.totalCount).toBe(1);
```

Comment:
Could be different when running concurrently, lets use toBeGreaterThanOrEqualTo or whatever it's called

Reply:
But maybe best to make sure the test is still doing a strong enough check, so let's also do
```ts
expect(res.body.response.filter((data) => data.userID === user1.id || data.userID === user2.id).length).toBe(1).
```
Quite likely we switch to --runInBand for everything in the future but lets keep this up for now.